docs/routers/OPNsense.md

```diff
@@ -0,0 +1,102 @@
+
+```
+(Clients) -> Pi-hole -> Upstream DNS Server
+```
+
+#### IPv4
+
+Go to **Services** --> **ISC DHCPv4** and click on the interface that you want to modify.
+Under **DNS servers** enter your Pi-hole IPv4 address into the first line.
+Leave the second line blank unless you have another Pi-hole host running.
+
+![Screenshot of OPNsense DHCPv4 Settings](../images/routers/opnsense-dhcpv4.png)
+
+#### IPv6
+
+
+Go to **Interfaces** and click on the interface that you want to modify.
+Under **Manual configuration** check "Allow manual adjustment of DHCPv6 and Router Advertisements"
+
+![Screenshot of OPNsense IPv6 manual RA Settings](../images/routers/opnsense-ipv6-manual-ra.png)
+
+Now go to **Services** --> **Router Advertisement** and click on the interface that you want to modify.
+Under **DNS servers** enter your Pi-Holes IPv6 address into the first line.
+Leave the second line blank unless you have another Pi-hole host running.
+
+![Screenshot of OPNsense IPv6 RA Settings](../images/routers/opensense-ra.png)
+
+Beware: Link local IPv6 are not reachable from other VLANs.
+
+
+All client devices on your network should now automatically be configured with a single IPv4 and single IPv6 address for DNS resolution. If you don't see your client devices update quickly enough, you *can* toggle Wi-Fi off/on or disconnect the Ethernet cable for a couple of seconds and then reconnect it.
+
+### Method 2: Using your Pi-hole as DNS server for your OPNsense
+
+If you'd rather have your OPNsense forward all DNS traffic to Pi-hole, you can complete this *alternative* step.
+Doing so will have DNS queries taking the following path:
+
+```
+(Clients) -> OPNsense -> Pi-hole -> Upstream DNS Server
+```
+
+!!! warning
+    Keep in mind that if you are using your OPNsense IP address as an upstream resolver for your Pi-hole together with this configuration, ***this will cause a resolver loop***. Make sure to have another upstream DNS server(s) set in your Pi-hole's configuration, or use Pi-hole as [your own recursive DNS server with Unbound](https://docs.pi-hole.net/guides/dns/unbound/?h=unbound#configure-unbound).
+
+1. In Pi-hole, navigate to `Settings -> DNS` and ensure you have at least one external upstream DNS server enabled.
+2. In OPNsense navigate to `Settings -> General -> Networking`.
+3. Under DNS Servers, enter the IPv4 and or IPv6 address of your Pi-hole host, and set the gateway to your WAN interface.
+4. Uncheck `Allow DNS server list to the overridden by DHCP/PPP on WAN`.
+5. Click Save.
+
+![Screenshot of OPNsense DNS WAN Settings](../images/routers/opnsense-dns-wan.png)
+
+### Block other DNS servers (optional)
+
+Some clients have hardcoded DNS settings. To prevent them from circumventing our Pi-Hole, we can optionally block all DNS request that are not coming from our Pi-Hole.
+!!! warning
+   If your device does not have any fallback from the hardcoded DNS settings, this could break its DNS.
```

Comment:
```suggestion
    If your device does not have any fallback from the hardcoded DNS settings, this could break its DNS.
```

I think you need 4 spaces to get the text into the warning box